From ab0bb5c2f810d75c61da91ed61d1278d1f1c2400 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 11 Jan 2021 22:49:01 +0100
Subject: [PATCH 5/5] rerun generate_llvm_build.py

From repository base, run:
python third_party/systemlibs/generate_llvm_build.py > third_party/systemlibs/llvm.BUILD
---
 third_party/systemlibs/llvm.BUILD | 2395 +++++++++++++++++++++++++++++
 1 file changed, 2395 insertions(+)
 create mode 100644 third_party/systemlibs/llvm.BUILD

diff --git a/third_party/systemlibs/llvm.BUILD b/third_party/systemlibs/llvm.BUILD
new file mode 100644
index 00000000000..f0c5120e908
--- /dev/null
+++ b/third_party/systemlibs/llvm.BUILD
@@ -0,0 +1,2395 @@
+
+# Bazel BUILD file for system LLVM
+#
+# Auto-generated by generate_llvm_build.py
+
+licenses(["notice"])
+
+filegroup(
+    name = "LICENSE.TXT",
+    visibility = ["//visibility:public"],
+)
+
+load(
+    "@org_tensorflow//third_party/llvm:llvm.bzl",
+    "cmake_var_string",
+    "expand_cmake_vars",
+    "gentbl",
+    "llvm_all_cmake_vars",
+    "llvm_copts",
+    "llvm_defines",
+    "llvm_linkopts",
+    "llvm_support_platform_specific_srcs_glob",
+)
+
+# Binary targets used by Tensorflow.
+genrule(
+    name = "lnllvmtblgenlink",
+    outs = ["llvmtblgenlink"],
+    cmd = "ln -s $$(which llvm-tblgen) $@",
+)
+
+sh_binary(
+    name = "llvm-tblgen",
+    srcs = ["llvmtblgenlink"],
+    visibility = ["//visibility:public"],
+)
+
+genrule(
+    name = "lnfilecheck",
+    outs = ["filechecklink"],
+    cmd = "ln -s $$(which FileCheck) $@",
+)
+
+sh_binary(
+    name = "FileCheck",
+    srcs = ["filechecklink"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+   name = "config",
+   defines = llvm_defines,
+   visibility = ["//visibility:public"],
+)
+
+cc_library(
+   name = "x86_defs",
+   visibility = ["//visibility:public"],
+   deps = [
+       ":config",
+   ]
+)
+
+
+cc_library(
+    name = "tblgen",
+    linkopts = ["-lLLVMtblgen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":TableGen",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "FileCheckLib",
+    linkopts = ["-lLLVMFileCheckLib"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+    ],
+)
+
+cc_library(
+    name = "AllTargetsCodeGens",
+    linkopts = ["-lLLVMAllTargetsCodeGens"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "AArch64CodeGen",
+        "AMDGPUCodeGen",
+        "ARMCodeGen",
+        "NVPTXCodeGen",
+        "PowerPCCodeGen",
+        "SystemZCodeGen",
+        "X86CodeGen",
+    ],
+)
+
+cc_library(
+    name = "Analysis",
+    linkopts = ["-lLLVMAnalysis"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":Core",
+        ":Object",
+        ":ProfileData",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64AsmParser",
+    linkopts = ["-lLLVMAArch64AsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AArch64Desc",
+        ":AArch64Info",
+        ":AArch64Utils",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64CodeGen",
+    linkopts = ["-lLLVMAArch64CodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AArch64Desc",
+        ":AArch64Info",
+        ":AArch64Utils",
+        ":Analysis",
+        ":AsmPrinter",
+        ":CFGuard",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64Desc",
+    linkopts = ["-lLLVMAArch64Desc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AArch64CommonTableGen",
+        ":AArch64Info",
+        ":AArch64Utils",
+        ":BinaryFormat",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64Disassembler",
+    linkopts = ["-lLLVMAArch64Disassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AArch64Desc",
+        ":AArch64Info",
+        ":AArch64Utils",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64Info",
+    linkopts = ["-lLLVMAArch64Info"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":CodeGen",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AArch64Utils",
+    linkopts = ["-lLLVMAArch64Utils"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AArch64CommonTableGen",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUAsmParser",
+    linkopts = ["-lLLVMAMDGPUAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUDesc",
+        ":AMDGPUInfo",
+        ":AMDGPUUtils",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUCodeGen",
+    linkopts = ["-lLLVMAMDGPUCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUDesc",
+        ":AMDGPUInfo",
+        ":AMDGPUUtils",
+        ":Analysis",
+        ":AsmPrinter",
+        ":BinaryFormat",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":IPO",
+        ":MC",
+        ":MIRParser",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":Vectorize",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUDesc",
+    linkopts = ["-lLLVMAMDGPUDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUInfo",
+        ":AMDGPUUtils",
+        ":BinaryFormat",
+        ":Core",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUDisassembler",
+    linkopts = ["-lLLVMAMDGPUDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUDesc",
+        ":AMDGPUInfo",
+        ":AMDGPUUtils",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUInfo",
+    linkopts = ["-lLLVMAMDGPUInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUCommonTableGen",
+        ":Core",
+        ":Support",
+        ":config",
+        ":r600_target_gen",
+    ],
+)
+
+cc_library(
+    name = "AMDGPUUtils",
+    linkopts = ["-lLLVMAMDGPUUtils"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AMDGPUCommonTableGen",
+        ":BinaryFormat",
+        ":Core",
+        ":MC",
+        ":Support",
+        ":config",
+        ":r600_target_gen",
+    ],
+)
+
+cc_library(
+    name = "ARCCodeGen",
+    linkopts = ["-lLLVMARCCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARCDesc",
+        ":ARCInfo",
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARCDesc",
+    linkopts = ["-lLLVMARCDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARCInfo",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARCDisassembler",
+    linkopts = ["-lLLVMARCDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARCInfo",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARCInfo",
+    linkopts = ["-lLLVMARCInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMAsmParser",
+    linkopts = ["-lLLVMARMAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMDesc",
+        ":ARMInfo",
+        ":ARMUtils",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMCodeGen",
+    linkopts = ["-lLLVMARMCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMDesc",
+        ":ARMInfo",
+        ":ARMUtils",
+        ":Analysis",
+        ":AsmPrinter",
+        ":CFGuard",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMDesc",
+    linkopts = ["-lLLVMARMDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMCommonTableGen",
+        ":ARMInfo",
+        ":ARMUtils",
+        ":BinaryFormat",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMDisassembler",
+    linkopts = ["-lLLVMARMDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMDesc",
+        ":ARMInfo",
+        ":ARMUtils",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMInfo",
+    linkopts = ["-lLLVMARMInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMCommonTableGen",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ARMUtils",
+    linkopts = ["-lLLVMARMUtils"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":ARMCommonTableGen",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AVRAsmParser",
+    linkopts = ["-lLLVMAVRAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AVRDesc",
+        ":AVRInfo",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AVRCodeGen",
+    linkopts = ["-lLLVMAVRCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AVRDesc",
+        ":AVRInfo",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AVRDesc",
+    linkopts = ["-lLLVMAVRDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AVRInfo",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AVRDisassembler",
+    linkopts = ["-lLLVMAVRDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AVRInfo",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AVRInfo",
+    linkopts = ["-lLLVMAVRInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AggressiveInstCombine",
+    linkopts = ["-lLLVMAggressiveInstCombine"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AsmParser",
+    linkopts = ["-lLLVMAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "AsmPrinter",
+    linkopts = ["-lLLVMAsmPrinter"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":BinaryFormat",
+        ":CodeGen",
+        ":Core",
+        ":DebugInfoCodeView",
+        ":DebugInfoDWARF",
+        ":DebugInfoMSF",
+        ":MC",
+        ":MCParser",
+        ":Remarks",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BPFAsmParser",
+    linkopts = ["-lLLVMBPFAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BPFDesc",
+        ":BPFInfo",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BPFCodeGen",
+    linkopts = ["-lLLVMBPFCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmPrinter",
+        ":BPFDesc",
+        ":BPFInfo",
+        ":CodeGen",
+        ":Core",
+        ":IPO",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BPFDesc",
+    linkopts = ["-lLLVMBPFDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BPFInfo",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BPFDisassembler",
+    linkopts = ["-lLLVMBPFDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BPFInfo",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BPFInfo",
+    linkopts = ["-lLLVMBPFInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BinaryFormat",
+    linkopts = ["-lLLVMBinaryFormat"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BitReader",
+    linkopts = ["-lLLVMBitReader"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BitstreamReader",
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BitWriter",
+    linkopts = ["-lLLVMBitWriter"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":MC",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "BitstreamReader",
+    linkopts = ["-lLLVMBitstreamReader"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "CFGuard",
+    linkopts = ["-lLLVMCFGuard"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "CSKYCodeGen",
+    linkopts = ["-lLLVMCSKYCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":CSKYInfo",
+        ":CodeGen",
+        ":Core",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "CSKYInfo",
+    linkopts = ["-lLLVMCSKYInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "CodeGen",
+    linkopts = ["-lLLVMCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":BitReader",
+        ":BitWriter",
+        ":Core",
+        ":Instrumentation",
+        ":MC",
+        ":ProfileData",
+        ":Scalar",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Core",
+    linkopts = ["-lLLVMCore"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":Remarks",
+        ":Support",
+        ":aarch64_enums_gen",
+        ":amdgcn_enums_gen",
+        ":arm_enums_gen",
+        ":bpf_enums_gen",
+        ":config",
+        ":hexagon_enums_gen",
+        ":mips_enums_gen",
+        ":nvvm_enums_gen",
+        ":ppc_enums_gen",
+        ":r600_enums_gen",
+        ":riscv_enums_gen",
+        ":s390_enums_gen",
+        ":wasm_enums_gen",
+        ":x86_enums_gen",
+        ":xcore_enums_gen",
+    ],
+)
+
+cc_library(
+    name = "Coroutines",
+    linkopts = ["-lLLVMCoroutines"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":IPO",
+        ":Scalar",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Coverage",
+    linkopts = ["-lLLVMCoverage"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Object",
+        ":ProfileData",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DWARFLinker",
+    linkopts = ["-lLLVMDWARFLinker"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmPrinter",
+        ":CodeGen",
+        ":DebugInfoDWARF",
+        ":MC",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DebugInfoCodeView",
+    linkopts = ["-lLLVMDebugInfoCodeView"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":DebugInfoMSF",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DebugInfoDWARF",
+    linkopts = ["-lLLVMDebugInfoDWARF"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":MC",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DebugInfoGSYM",
+    linkopts = ["-lLLVMDebugInfoGSYM"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":DebugInfoDWARF",
+        ":MC",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DebugInfoMSF",
+    linkopts = ["-lLLVMDebugInfoMSF"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DebugInfoPDB",
+    linkopts = ["-lLLVMDebugInfoPDB"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":DebugInfoCodeView",
+        ":DebugInfoMSF",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Demangle",
+    linkopts = ["-lLLVMDemangle"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "DlltoolDriver",
+    linkopts = ["-lLLVMDlltoolDriver"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Object",
+        ":Option",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ExecutionEngine",
+    linkopts = ["-lLLVMExecutionEngine"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":MC",
+        ":Object",
+        ":RuntimeDyld",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Extensions",
+    linkopts = ["-lLLVMExtensions"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "FrontendOpenMP",
+    linkopts = ["-lLLVMFrontendOpenMP"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+        ":omp_gen",
+        ":omp_gen_impl",
+    ],
+)
+
+cc_library(
+    name = "FuzzMutate",
+    linkopts = ["-lLLVMFuzzMutate"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":BitReader",
+        ":BitWriter",
+        ":Core",
+        ":Scalar",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "GlobalISel",
+    linkopts = ["-lLLVMGlobalISel"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HelloNew",
+    linkopts = ["-lLLVMHelloNew"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HexagonAsmParser",
+    linkopts = ["-lLLVMHexagonAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":HexagonDesc",
+        ":HexagonInfo",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HexagonCodeGen",
+    linkopts = ["-lLLVMHexagonCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":HexagonAsmParser",
+        ":HexagonDesc",
+        ":HexagonInfo",
+        ":IPO",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HexagonDesc",
+    linkopts = ["-lLLVMHexagonDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":HexagonInfo",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HexagonDisassembler",
+    linkopts = ["-lLLVMHexagonDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":HexagonDesc",
+        ":HexagonInfo",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "HexagonInfo",
+    linkopts = ["-lLLVMHexagonInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "IPO",
+    linkopts = ["-lLLVMIPO"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AggressiveInstCombine",
+        ":Analysis",
+        ":BitReader",
+        ":BitWriter",
+        ":Core",
+        ":FrontendOpenMP",
+        ":IRReader",
+        ":InstCombine",
+        ":Instrumentation",
+        ":Linker",
+        ":Object",
+        ":ProfileData",
+        ":Scalar",
+        ":Support",
+        ":TransformUtils",
+        ":Vectorize",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "IRReader",
+    linkopts = ["-lLLVMIRReader"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmParser",
+        ":BitReader",
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "InstCombine",
+    linkopts = ["-lLLVMInstCombine"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":InstCombineTableGen",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Instrumentation",
+    linkopts = ["-lLLVMInstrumentation"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":MC",
+        ":ProfileData",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "InterfaceStub",
+    linkopts = ["-lLLVMInterfaceStub"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Interpreter",
+    linkopts = ["-lLLVMInterpreter"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":CodeGen",
+        ":Core",
+        ":ExecutionEngine",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "JITLink",
+    linkopts = ["-lLLVMJITLink"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LTO",
+    linkopts = ["-lLLVMLTO"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AggressiveInstCombine",
+        ":Analysis",
+        ":BinaryFormat",
+        ":BitReader",
+        ":BitWriter",
+        ":CodeGen",
+        ":Core",
+        ":Extensions",
+        ":IPO",
+        ":InstCombine",
+        ":Linker",
+        ":MC",
+        ":ObjCARC",
+        ":Object",
+        ":Passes",
+        ":Remarks",
+        ":Scalar",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LanaiAsmParser",
+    linkopts = ["-lLLVMLanaiAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":LanaiDesc",
+        ":LanaiInfo",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LanaiCodeGen",
+    linkopts = ["-lLLVMLanaiCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":LanaiAsmParser",
+        ":LanaiDesc",
+        ":LanaiInfo",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LanaiDesc",
+    linkopts = ["-lLLVMLanaiDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":LanaiInfo",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LanaiDisassembler",
+    linkopts = ["-lLLVMLanaiDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":LanaiDesc",
+        ":LanaiInfo",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LanaiInfo",
+    linkopts = ["-lLLVMLanaiInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LibDriver",
+    linkopts = ["-lLLVMLibDriver"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":BitReader",
+        ":Object",
+        ":Option",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "LineEditor",
+    linkopts = ["-lLLVMLineEditor"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Linker",
+    linkopts = ["-lLLVMLinker"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MC",
+    linkopts = ["-lLLVMMC"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":DebugInfoCodeView",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MCA",
+    linkopts = ["-lLLVMMCA"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MCDisassembler",
+    linkopts = ["-lLLVMMCDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MCJIT",
+    linkopts = ["-lLLVMMCJIT"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":ExecutionEngine",
+        ":Object",
+        ":RuntimeDyld",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MCParser",
+    linkopts = ["-lLLVMMCParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MIRParser",
+    linkopts = ["-lLLVMMIRParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmParser",
+        ":BinaryFormat",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MSP430AsmParser",
+    linkopts = ["-lLLVMMSP430AsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":MSP430Desc",
+        ":MSP430Info",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MSP430CodeGen",
+    linkopts = ["-lLLVMMSP430CodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":MSP430Desc",
+        ":MSP430Info",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MSP430Desc",
+    linkopts = ["-lLLVMMSP430Desc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MSP430Info",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MSP430Disassembler",
+    linkopts = ["-lLLVMMSP430Disassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":MSP430Info",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MSP430Info",
+    linkopts = ["-lLLVMMSP430Info"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MipsAsmParser",
+    linkopts = ["-lLLVMMipsAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":MipsDesc",
+        ":MipsInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MipsCodeGen",
+    linkopts = ["-lLLVMMipsCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":MipsDesc",
+        ":MipsInfo",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MipsDesc",
+    linkopts = ["-lLLVMMipsDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MipsInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MipsDisassembler",
+    linkopts = ["-lLLVMMipsDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":MipsInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "MipsInfo",
+    linkopts = ["-lLLVMMipsInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "NVPTXCodeGen",
+    linkopts = ["-lLLVMNVPTXCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":IPO",
+        ":MC",
+        ":NVPTXDesc",
+        ":NVPTXInfo",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":Vectorize",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "NVPTXDesc",
+    linkopts = ["-lLLVMNVPTXDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":NVPTXCommonTableGen",
+        ":NVPTXInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "NVPTXInfo",
+    linkopts = ["-lLLVMNVPTXInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":NVPTXCommonTableGen",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ObjCARC",
+    linkopts = ["-lLLVMObjCARC"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Object",
+    linkopts = ["-lLLVMObject"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":BitReader",
+        ":Core",
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":TextAPI",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ObjectYAML",
+    linkopts = ["-lLLVMObjectYAML"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":DebugInfoCodeView",
+        ":MC",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Option",
+    linkopts = ["-lLLVMOption"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "OrcError",
+    linkopts = ["-lLLVMOrcError"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "OrcJIT",
+    linkopts = ["-lLLVMOrcJIT"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":ExecutionEngine",
+        ":JITLink",
+        ":MC",
+        ":Object",
+        ":OrcError",
+        ":Passes",
+        ":RuntimeDyld",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Passes",
+    linkopts = ["-lLLVMPasses"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AggressiveInstCombine",
+        ":Analysis",
+        ":CodeGen",
+        ":Core",
+        ":Coroutines",
+        ":HelloNew",
+        ":IPO",
+        ":InstCombine",
+        ":Instrumentation",
+        ":ObjCARC",
+        ":Scalar",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":Vectorize",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "PowerPCAsmParser",
+    linkopts = ["-lLLVMPowerPCAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":PowerPCDesc",
+        ":PowerPCInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "PowerPCCodeGen",
+    linkopts = ["-lLLVMPowerPCCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":PowerPCDesc",
+        ":PowerPCInfo",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "PowerPCDesc",
+    linkopts = ["-lLLVMPowerPCDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":MC",
+        ":PowerPCCommonTableGen",
+        ":PowerPCInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "PowerPCDisassembler",
+    linkopts = ["-lLLVMPowerPCDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":PowerPCInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "PowerPCInfo",
+    linkopts = ["-lLLVMPowerPCInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":PowerPCCommonTableGen",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "ProfileData",
+    linkopts = ["-lLLVMProfileData"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":Demangle",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVAsmParser",
+    linkopts = ["-lLLVMRISCVAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":RISCVDesc",
+        ":RISCVInfo",
+        ":RISCVUtils",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVCodeGen",
+    linkopts = ["-lLLVMRISCVCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":RISCVDesc",
+        ":RISCVInfo",
+        ":RISCVUtils",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVDesc",
+    linkopts = ["-lLLVMRISCVDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":RISCVInfo",
+        ":RISCVUtils",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVDisassembler",
+    linkopts = ["-lLLVMRISCVDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":RISCVInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVInfo",
+    linkopts = ["-lLLVMRISCVInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RISCVUtils",
+    linkopts = ["-lLLVMRISCVUtils"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Remarks",
+    linkopts = ["-lLLVMRemarks"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BitstreamReader",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "RuntimeDyld",
+    linkopts = ["-lLLVMRuntimeDyld"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Core",
+        ":MC",
+        ":MCDisassembler",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Scalar",
+    linkopts = ["-lLLVMScalar"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AggressiveInstCombine",
+        ":Analysis",
+        ":Core",
+        ":InstCombine",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SelectionDAG",
+    linkopts = ["-lLLVMSelectionDAG"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SparcAsmParser",
+    linkopts = ["-lLLVMSparcAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":SparcDesc",
+        ":SparcInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SparcCodeGen",
+    linkopts = ["-lLLVMSparcCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":SparcDesc",
+        ":SparcInfo",
+        ":Support",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SparcDesc",
+    linkopts = ["-lLLVMSparcDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":SparcInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SparcDisassembler",
+    linkopts = ["-lLLVMSparcDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":SparcInfo",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SparcInfo",
+    linkopts = ["-lLLVMSparcInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Support",
+    linkopts = ["-lLLVMSupport"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Demangle",
+        ":config",
+        "@zlib",
+    ],
+)
+
+cc_library(
+    name = "Symbolize",
+    linkopts = ["-lLLVMSymbolize"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":DebugInfoDWARF",
+        ":DebugInfoPDB",
+        ":Demangle",
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SystemZAsmParser",
+    linkopts = ["-lLLVMSystemZAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":SystemZDesc",
+        ":SystemZInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SystemZCodeGen",
+    linkopts = ["-lLLVMSystemZCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":SystemZDesc",
+        ":SystemZInfo",
+        ":Target",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SystemZDesc",
+    linkopts = ["-lLLVMSystemZDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":SystemZCommonTableGen",
+        ":SystemZInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SystemZDisassembler",
+    linkopts = ["-lLLVMSystemZDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":SystemZDesc",
+        ":SystemZInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "SystemZInfo",
+    linkopts = ["-lLLVMSystemZInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":SystemZCommonTableGen",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "TableGen",
+    linkopts = ["-lLLVMTableGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Target",
+    linkopts = ["-lLLVMTarget"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":MC",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "TestingSupport",
+    linkopts = ["-lLLVMTestingSupport"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "TextAPI",
+    linkopts = ["-lLLVMTextAPI"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "TransformUtils",
+    linkopts = ["-lLLVMTransformUtils"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "VEAsmParser",
+    linkopts = ["-lLLVMVEAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":VEDesc",
+        ":VEInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "VECodeGen",
+    linkopts = ["-lLLVMVECodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":VEDesc",
+        ":VEInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "VEDesc",
+    linkopts = ["-lLLVMVEDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":VEInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "VEDisassembler",
+    linkopts = ["-lLLVMVEDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":Support",
+        ":VEInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "VEInfo",
+    linkopts = ["-lLLVMVEInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "Vectorize",
+    linkopts = ["-lLLVMVectorize"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":Core",
+        ":Scalar",
+        ":Support",
+        ":TransformUtils",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WebAssemblyAsmParser",
+    linkopts = ["-lLLVMWebAssemblyAsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":WebAssemblyInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WebAssemblyCodeGen",
+    linkopts = ["-lLLVMWebAssemblyCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":BinaryFormat",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":Scalar",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":WebAssemblyDesc",
+        ":WebAssemblyInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WebAssemblyDesc",
+    linkopts = ["-lLLVMWebAssemblyDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":WebAssemblyInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WebAssemblyDisassembler",
+    linkopts = ["-lLLVMWebAssemblyDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":WebAssemblyDesc",
+        ":WebAssemblyInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WebAssemblyInfo",
+    linkopts = ["-lLLVMWebAssemblyInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "WindowsManifest",
+    linkopts = ["-lLLVMWindowsManifest"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "X86AsmParser",
+    linkopts = ["-lLLVMX86AsmParser"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":MCParser",
+        ":Support",
+        ":X86Desc",
+        ":X86Info",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "X86CodeGen",
+    linkopts = ["-lLLVMX86CodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CFGuard",
+        ":CodeGen",
+        ":Core",
+        ":GlobalISel",
+        ":MC",
+        ":ProfileData",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":X86Desc",
+        ":X86Info",
+        ":config",
+        ":x86_defs",
+    ],
+)
+
+cc_library(
+    name = "X86Desc",
+    linkopts = ["-lLLVMX86Desc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":BinaryFormat",
+        ":MC",
+        ":MCDisassembler",
+        ":Support",
+        ":X86Info",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "X86Disassembler",
+    linkopts = ["-lLLVMX86Disassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":Support",
+        ":X86Info",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "X86Info",
+    linkopts = ["-lLLVMX86Info"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":X86CommonTableGen",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "XCoreCodeGen",
+    linkopts = ["-lLLVMXCoreCodeGen"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Analysis",
+        ":AsmPrinter",
+        ":CodeGen",
+        ":Core",
+        ":MC",
+        ":SelectionDAG",
+        ":Support",
+        ":Target",
+        ":TransformUtils",
+        ":XCoreDesc",
+        ":XCoreInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "XCoreDesc",
+    linkopts = ["-lLLVMXCoreDesc"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MC",
+        ":Support",
+        ":XCoreInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "XCoreDisassembler",
+    linkopts = ["-lLLVMXCoreDisassembler"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":MCDisassembler",
+        ":Support",
+        ":XCoreInfo",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "XCoreInfo",
+    linkopts = ["-lLLVMXCoreInfo"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Support",
+        ":config",
+    ],
+)
+
+cc_library(
+    name = "XRay",
+    linkopts = ["-lLLVMXRay"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":Object",
+        ":Support",
+        ":config",
+    ],
+)
-- 
2.29.2.windows.3

